core/src/components/datetime/datetime.tsx

```diff
@@ -886,17 +841,35 @@ export class Datetime implements ComponentInterface {
         const root = this.el!.shadowRoot!;
 
         /**
-         * Get the element that is in the center of the calendar body.
-         * This will be an element inside of the active month.
+         * Get the elements that are in the center of the calendar body.
+         * The topmost of these will be elements inside of the active month.
+         *
+         * We filter out popovers to exclude the time picker, if present,
+         * since it sits on top of the entire screen. This handles the
+         * time picker being open when the datetime's value is changed
+         * programmatically, for example.
+         */
+        const elementsAtCenter = root
```

Comment:
FYI This might be impacted by https://github.com/ionic-team/ionic-framework/pull/27917. I found a way to accomplish this without relying on `elementFromPoint` which seemed to have a lot of edge cases.

Reply:
Ah, good callout. I think your change might avoid the popover issue completely 👀

Reply:
Got your changes merged in 👍